### PR TITLE
Change heuristics for showing missing `.env` error

### LINKED ISSUE #9479: Spurious warning in dashboard tests
Consider the following test run https://github.com/enso-org/enso/actions/runs/8328687799/job/22789224099?pr=9406#step:8:206

The following is printed to the error output:
```
Could not load `.env` file; disabling cloud backend.
[Error: ENOENT: no such file or directory, open '/runner/_work/enso/enso/app/ide-desktop/.env'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/runner/_work/enso/enso/app/ide-desktop/.env'
}
```

I think this is an expected situation (why would I need backend to run gui tests?) — having some faux error being printed makes it confusing when looking for the actual errors.

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
- Fix #9479

### Important Notes
None

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] ~~Unit tests have been written where possible.~~
  - [x] ~~If GUI codebase was changed, the GUI was tested when built using `./run ide build`.~~

### CODE REVIEW

File: app/ide-desktop/lib/common/src/appConfig.js [hunk 1]
```diff
@@ -36,12 +36,19 @@ export async function readEnvironmentFromFile() {
...
+            key.replace(/^proccess[.]env[.]/, '')
+        )
+        let isKeyMissing = false
+        for (const key of expectedKeys) {
+            if (!(key in process.env)) {
+                isKeyMissing = true
+                break
+            }
+        }
+        if (isKeyMissing) {
```
Comment (MrFlashAccount) [inline app/ide-desktop/lib/common/src/appConfig.js]: Maybe we can also print which keys are missing though?

**Comment by MrFlashAccount** [comment]
Do we use these configurations for unit tests only?

**Comment by somebody1234** [comment]
these configurations are used for all builds

File: app/gui2/vitest.config.ts [hunk 2]
```diff
@@ -1,7 +1,10 @@
+import { loadTestEnvironmentVariables } from 'enso-common/src/appConfig'
 import { fileURLToPath } from 'node:url'
 import { configDefaults, defineConfig, mergeConfig } from 'vitest/config'
 import viteConfig from './vite.config'
 
+loadTestEnvironmentVariables()
+
```
Comment (farmaazon) [inline app/gui2/vitest.config.ts]: Why do we need it in GUI2 tests too?

Comment (somebody1234) [reply app/gui2/vitest.config.ts]: good question... GUI2 tests don't load the dashboard at all right?

Comment (farmaazon) [reply app/gui2/vitest.config.ts]: I'm not aware of any unit test that would. Only E2E tests, maybe.

**Comment by MrFlashAccount** [comment]
> these configurations are used for all builds

I meant vitest.config.ts, sorry for confusion.

**Comment by somebody1234** [comment]
ah, yeah:
- `vitest` (gui) is for gui unit tests only iirc?
- `vite` (gui) is for the regular app
- `vite.test` (dashboard) is for dashboard unit tests only.
- `vite` (dashboard) is for both unit tests and the regular app

**Comment by MrFlashAccount** [comment]
I think, if we have to specify env variables for unit tests - we're doing something wrong, and the problem is not with the env, but with our code (maybe bacause our env variable spread across the app instead of staying located in the single place?)

**Comment by somebody1234** [comment]
@MrFlashAccount @PabloBuchu @indiv0 mind reviewing?

**Review by MrFlashAccount — APPROVED** [review]

**Pull request merged**